docs/Versioning.md

```diff
@@ -0,0 +1,73 @@
+# ML-Agents Versioning
+
+## Context
+As the ML-Agents project evolves into a more mature product, we want to communicate the process
+we use to version our packages and the data that flows into, through, and out of them clearly.
+Our project now has four packages (1 Unity, 3 Python) along with artifacts that are produced as
+well as consumed.  This document covers the versioning for these packages and artifacts.
+
+## Packages
+All of the software packages, and their generated artifacts will be versioned.  Any automation
+tools will not be versioned.
+
+### Unity package
+Package name: com.unity.ml-agents
+- Versioned following [Semantic Versioning Guidelines](https://www.semver.org)
+- This package produces one artifact, the `.demo` files.  These files will have integer
+    versioning. This means their version will increment by 1 at each change.  The
+    com.unity.ml-agents package must be backward compatible with version changes
+    that occur between minor versions.  For example, consider that com.unity.ml-agents
+    is at version 1.0.0 and the demo files are at version one.  If the demo files
```

Comment:
Do we ever read .demo files in the C# package or only write? I thought we generated them in C# and consumed them in Python.

Reply:
The only time we read them is in the Demonstration Drawer, and that's just summary stats.

Reply:
you are correct

Reply:
i used the NN files as an example instead.